src/diffcalc_API/utils.py

```diff
@@ -0,0 +1,13 @@
+import asyncio
+from functools import partial, wraps
+
+
+def async_wrap(func):
```

Comment:
Not sure what this is for, could you explain how you came up with it?

Reply:
I just took this from here: https://dev.to/0xbf/turn-sync-function-to-async-python-tips-58nn

The main purpose of this wrapper is just to make synchronous code run asynchronously on the event loop.